src/scenes/PosterPage/PosterMintButton.tsx

```diff
@@ -0,0 +1,92 @@
+import { Web3Provider } from '@ethersproject/providers';
+import { useWeb3React } from '@web3-react/core';
+import Button from 'components/core/Button/Button';
+import GalleryLink from 'components/core/GalleryLink/GalleryLink';
+import Spacer from 'components/core/Spacer/Spacer';
+import ErrorText from 'components/core/Text/ErrorText';
+import { BaseM } from 'components/core/Text/Text';
+import { NFT_TOKEN_ID } from 'constants/poster';
+import { useModalActions } from 'contexts/modal/ModalContext';
+import { useMintPosterContract } from 'hooks/useContract';
+import useMintContract from 'hooks/useMintContract';
+import useWalletModal from 'hooks/useWalletModal';
+import { useCallback, useEffect } from 'react';
+import { TransactionStatus } from 'constants/transaction';
+import styled from 'styled-components';
+
+// TODO: Might be a great idea to encapsulated this method from mint page
+export default function PosterMintButton() {
+  const { active } = useWeb3React<Web3Provider>();
+
+  const showWalletModal = useWalletModal();
+  const { hideModal } = useModalActions();
+
+  const tokenId = NFT_TOKEN_ID;
+
+  const contract = useMintPosterContract();
+  const {
+    transactionHash,
+    transactionStatus,
+    buttonText,
+    error,
+    handleMintButtonClick,
+  } = useMintContract({
+    contract,
+    tokenId,
+  });
+
+  useEffect(() => {
+    if (active) {
+      hideModal();
+    }
+  }, [active, hideModal]);
+
+  const handleConnectWalletButtonClick = useCallback(() => {
+    if (!active) {
+      showWalletModal();
+    }
+  }, [active, showWalletModal]);
+
+  const handleOnClick = () => {
+    active ? handleMintButtonClick() : handleConnectWalletButtonClick();
+  };
+
+  return (
+    <>
+      <StyledButton onClick={handleOnClick} text={buttonText}></StyledButton>
```

Comment:
can we disable the mint button while the transaction status is PENDING?
that way users can't accidentally trigger another transaction while they're waiting

![Screen Shot 2022-06-01 at 20 39 00](https://user-images.githubusercontent.com/80802871/171396566-58bad6ea-594b-4cea-8ed3-36ee83ede36b.png)